### PR TITLE
Add support for restrictedToMinimumLevel

### PR DESCRIPTION
**Pull request by Slushnas**
This is a small change that allows passing the restrictedToMinimumLevel configuration argument up the chain to Serilog which should close out issue #1.

### CODE REVIEW

File: src/Serilog.Sinks.Syslog/SyslogLoggerConfigurationExtensions.cs [hunk 1]
```diff
@@ -28,10 +29,12 @@ public static class SyslogLoggerConfigurationExtensions
         /// <param name="loggerSinkConfig">The logger configuration</param>
         /// <param name="facility">The category of the application</param>
         /// <param name="outputTemplate">A message template describing the output messages
+        /// <param name="restrictedToMinimumLevel">The minimum level for events passed through the sink</param>
         /// <seealso cref="https://github.com/serilog/serilog/wiki/Formatting-Output"/>
         /// </param>
         public static LoggerConfiguration LocalSyslog(this LoggerSinkConfiguration loggerSinkConfig,
-            Facility facility = Facility.Local0, string outputTemplate = null)
+            Facility facility = Facility.Local0, string outputTemplate = null,
+            LogEventLevel restrictedToMinimumLevel = LogEventLevel.Verbose)
```
Comment (colin-anderson) [inline src/Serilog.Sinks.Syslog/SyslogLoggerConfigurationExtensions.cs]:
```suggestion
            LogEventLevel restrictedToMinimumLevel = LevelAlias.Minimum)
```

File: src/Serilog.Sinks.Syslog/SyslogLoggerConfigurationExtensions.cs [hunk 2]
```diff
@@ -55,11 +58,13 @@ public static class SyslogLoggerConfigurationExtensions
...
         /// <param name="facility">The category of the application</param>
         /// <param name="outputTemplate">A message template describing the output messages
+        /// <param name="restrictedToMinimumLevel">The minimum level for events passed through the sink</param>
         /// <seealso cref="https://github.com/serilog/serilog/wiki/Formatting-Output"/>
         /// </param>
         public static LoggerConfiguration UdpSyslog(this LoggerSinkConfiguration loggerSinkConfig,
             string host, int port = 514, string appName = null, SyslogFormat format = SyslogFormat.RFC3164,
-            Facility facility = Facility.Local0, string outputTemplate = null)
+            Facility facility = Facility.Local0, string outputTemplate = null,
+            LogEventLevel restrictedToMinimumLevel = LogEventLevel.Verbose)
```
Comment (colin-anderson) [inline src/Serilog.Sinks.Syslog/SyslogLoggerConfigurationExtensions.cs]:
```suggestion
            LogEventLevel restrictedToMinimumLevel = LevelAlias.Minimum)
```

File: src/Serilog.Sinks.Syslog/SyslogLoggerConfigurationExtensions.cs [hunk 3]
```diff
@@ -78,15 +83,16 @@ public static class SyslogLoggerConfigurationExtensions
         /// </summary>
         /// <param name="loggerSinkConfig">The logger configuration</param>
         /// <param name="config">Defines how to interact with the syslog server</param>
+        /// <param name="restrictedToMinimumLevel">The minimum level for events passed through the sink</param>
         public static LoggerConfiguration TcpSyslog(this LoggerSinkConfiguration loggerSinkConfig,
-            SyslogTcpConfig config)
+            SyslogTcpConfig config, LogEventLevel restricedToMinimumLevel = LogEventLevel.Verbose)
```
Comment (colin-anderson) [inline src/Serilog.Sinks.Syslog/SyslogLoggerConfigurationExtensions.cs]:
```suggestion
            SyslogTcpConfig config, LogEventLevel restricedToMinimumLevel = LevelAlias.Minimum)
```

File: src/Serilog.Sinks.Syslog/SyslogLoggerConfigurationExtensions.cs [hunk 4]
```diff
@@ -110,12 +116,14 @@ public static class SyslogLoggerConfigurationExtensions
...
         /// </param>
+        /// <param name="restrictedToMinimumLevel">The minimum level for events passed through the sink</param>
         public static LoggerConfiguration TcpSyslog(this LoggerSinkConfiguration loggerSinkConfig,
             string host, int port = 1468, string appName = null, FramingType framingType = FramingType.OCTET_COUNTING,
             SyslogFormat format = SyslogFormat.RFC5424, Facility facility = Facility.Local0,
             SslProtocols secureProtocols = SslProtocols.Tls12, ICertificateProvider certProvider = null,
             RemoteCertificateValidationCallback certValidationCallback = null,
-            string outputTemplate = null)
+            string outputTemplate = null,
+            LogEventLevel restrictedToMinimumLevel = LogEventLevel.Verbose)
```
Comment (colin-anderson) [inline src/Serilog.Sinks.Syslog/SyslogLoggerConfigurationExtensions.cs]:
```suggestion
            LogEventLevel restrictedToMinimumLevel = LevelAlias.Minimum)
```

File: src/Serilog.Sinks.Syslog/SyslogLoggerConfigurationExtensions.cs (hunk 1 above)
Comment (colin-anderson) [reply src/Serilog.Sinks.Syslog/SyslogLoggerConfigurationExtensions.cs]: Just to match the other Serilog sinks, can we change this to `LevelAlias.Minimum` instead? They currently mean the same thing, but it matches up with the other sinks.

**Comment by Slushnas** [comment]
I think that should do it.

**Comment by colin-anderson** [comment]
@Slushnas merged, thanks for your contribution!

Also, for future reference, GitHub now has a great code review feature where the reviewer can suggest simple changes inline - and the reviewee needs to do is check and hit the "Approve" button in the PR to make it so. As you added a new commit rather than using the Approve buttons, I just thought I'd mention it in case you weren't aware; GitHub doesn't make it a very discoverable feature, IMO, which is a shame :smile:

**Pull request merged**